include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1084,11 +1124,93 @@ struct __early_exit_find_or
 // parallel_find_or - sync pattern
 //------------------------------------------------------------------------
 
-// Base pattern for __parallel_or and __parallel_find. The execution depends on tag type _BrickTag.
+// Specialization for __parallel_or_tag
+template <typename _ExecutionPolicy, typename _Brick, typename... _Ranges>
+bool
+__parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec, _Brick __f,
```

Comment:
Essentially, did you "split" `__parallel_find_or` pattern into two with "OR search logic" and "first/last search logic".
In that cause I would propose to have different names for patterns: `__parallel_find_or `and `__parallel_find_first`.

Reply:
@MikeDvorskiy, how about `__parallel_find_or` and `__parallel_find_entry` ?

Reply:
the standard and cppreference uses the "first" word. 
As  variant
`__parallel_find_first`
and 
`__parallel_find_any`

Reply:
Fixed: `__parallel_find_or` has been renamed to `__parallel_find_first` and `__parallel_find_any`
And some staff too.